roles/agnosticv/tasks/generate_catalog_item_tasks.yml

```diff
@@ -17,14 +36,15 @@
     namespace: "{{ template_namespace }}"
   register: template_facts
 
-- name: Generate Catalog item
+- name: Generate Catalog item (template)
   template:
-    src: catalog_item.yml.j2
-    dest: "{{ output_dir }}/{{ account }}_{{ catalog_item }}_{{ stage }}.yml"
+    src: template.yml.j2
+    dest: "{{ output_dir }}/{{ account }}_{{ catalog_item }}_{{ stage }}_template.yml"
   vars:
     _name: "{{ catalog_item_name }}"
     current_resource_version: >-
-      {{ template_facts.resources[0].metadata.resourceVersion if template_facts.resources else '' }}
+      {{ template_facts.resources[0].metadata.resourceVersion
+      if template_facts.resources else '' }}
```

Comment:
I see this pattern of using `template` to create a file then another task to create the resource... It should be significantly faster to avoid the template file creation with `lookup('template', ...)` in the `k8s` task.